# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/pkg_patch/pkg_nasm.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# Copyright (C) 1998 - 2004 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- nasm-0.98.36/nasmlib.c~	Thu Feb  6 02:44:09 2003
+++ nasm-0.98.36/nasmlib.c	Sun Sep  7 23:23:29 2003
@@ -372,10 +372,8 @@
     if (posn >= r->stepsize * LAYERSIZ(r))
 	return 0;		/* Return 0 for undefined entries */
     while (r->layers > 0) {
-	ldiv_t l;
-	l = ldiv (posn, r->stepsize);
-	r = r->u.b.data[l.quot];
-	posn = l.rem;
+	r = r->u.b.data[posn/r->stepsize];
+	posn = posn%r->stepsize;
 	if (!r)
 	    return 0;		/* Return 0 for undefined entries */
     }
@@ -408,14 +406,12 @@
     result = r;
 
     while (r->layers > 0) {
-	ldiv_t l;
 	struct RAA **s;
-	l = ldiv (posn, r->stepsize);
-	s = &r->u.b.data[l.quot];
+	s = &r->u.b.data[posn / r->stepsize];
 	if (!*s)
 	    *s = real_raa_init (r->layers - 1);
 	r = *s;
-	posn = l.rem;
+	posn = posn % r->stepsize;
     }
 
     r->u.l.data[posn] = value;
